app/controllers/api/v2/accredited_body_training_provider_courses_controller.rb

```diff
@@ -0,0 +1,37 @@
+module API
+  module V2
+    class AccreditedBodyTrainingProviderCoursesController < ApplicationController
+      before_action :build_recruitment_cycle, :build_accredited_body, :build_training_provider
+
+      def index
+        authorize accredited_body, :can_list_training_providers?
+
+        courses = policy_scope(Course).where(provider_id: training_provider.id)
```

Comment:
Just an aside, not a blocker, but ... after running into all the slow Apply requests, I think we need to focus a bit more on performance. There are not includes being done here so it worries me a bit, we'll have to think of a generalised approach that can be ported to other endpoints.

Reply:
We should benchmark this stuff and get some facts to base that on. I've had a quick look at `SerializableCourse` I don't think we get N+1 anywhere, we just get a few more DB queries. Having 4 or 5 queries might be quicker than trying to include everything here (and of course it might not but it's probably negligible). Might be a good idea to have a maximum response time that we're aiming for on V3.